src/Lykke.Service.LoggingAdapter.Services/ShutdownManager.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Common;
+using Common.Log;
+using Lykke.Common.Log;
+using Lykke.Service.LoggingAdapter.Core.Services;
+
+namespace Lykke.Service.LoggingAdapter.Services
+{
+    // NOTE: Sometimes, shutdown process should be expressed explicitly. 
+    // If this is your case, use this class to manage shutdown.
+    // For example, sometimes some state should be saved only after all incoming message processing and 
+    // all periodical handler was stopped, and so on.
+
+    public class ShutdownManager : IShutdownManager
+    {
+        private readonly ILog _log;
+        private readonly IEnumerable<IStopable> _items;
+
+        public ShutdownManager(ILogFactory logFactory, IEnumerable<IStopable> items)
+        {
+            _log = logFactory.CreateLog(this);
+            _items = items;
+        }
+
+        public async Task StopAsync()
+        {
+            // TODO: Implement your shutdown logic here. Good idea is to log every step
+            foreach (var item in _items)
+            {
+                try
+                {
+                    item.Stop();
+                }
+                catch (Exception ex)
+                {
+                    _log.WriteWarning(nameof(StopAsync), null, $"Unable to stop {item.GetType().Name}", ex);
```

Comment:
SInce UseLykkeLogging is used, you could use new logging methods everywhere (Tace, Debug, Info, Warning, Error, Critical)